### PR TITLE
populate group list when doing token review

### LINKED ISSUE #9869: pull / push image with token got access denied with Harbor http_auth though user's group has the master role
**Expected behavior and actual behavior:**
User is not able to pull / push images with token after login (username is tokenreview$<username>), on Harbor configured with http_auth, though the group that the user belongs has master role.

**Steps to reproduce the problem:**
1. Add user's group (i.e. vsphere.local\\administrators) to the Harbor project

2. Login to Harbor registry via docker-crendential-vsphere as a user in the group above, and got the jwt-token:
docker-credential-vsphere login 192.168.123.2 --tlscacert /etc/docker/certs.d/192.168.123.2/ca.crt
Note that after the login, the user name will be tokenreview$<username>

3. run docker push command like the following failed:
root@sc-rdops-vm10-dhcp-33-182:~# docker push 192.168.123.2/test/busybox:latest
The push refers to a repository [192.168.123.2/test/busybox]
1da8e4c8d307: Preparing 
denied: requested access to the resource is denied

The image push works fine if using the native: docker login command, or add the user as a member directly to the project. The issue should be the tokenreview user handing for the case that only user's group has the needed privilege in the project.

**Versions:**
Please specify the versions of following systems.

- harbor version: 1.9.1

** logs **
Harbor core logs:

2019-11-13T21:57:56Z [DEBUG] [/core/service/token/token.go:35]: URL for token request: /service/token?account=tokenreview%24administrator%40vsphere.local&scope=repository%3Atest%2Fbusybox%3Apush%2Cpull&service=harbor-registry
2019-11-13T21:57:56Z [DEBUG] [/core/service/token/creator.go:202]: scopes: [repository:test/busybox:push,pull]
2019-11-13T21:57:56Z [DEBUG] [/core/service/token/authutils.go:47]: scopes: [repository:test/busybox:push,pull]
2019-11-13T21:57:56Z [DEBUG] [/core/service/token/authutils.go:94]: user: administrator@vsphere.local, access: &{repository  test/busybox []}
2019/11/13 21:57:56 ^[[1;44m[D] [server.go:2774] |    100.64.80.3|^[[42m 200 ^[[0m| 301.951378ms|   match|^[[44m GET     ^[[0m /service/token   r:/service/token^[[0m
2019-11-13T21:57:56Z [DEBUG] [/core/filter/security.go:227]: OIDC CLI modifier only handles request by docker CLI or helm CLI
2019-11-13T21:57:56Z [DEBUG] [/core/filter/security.go:302]: Auth proxy modifier only handles docker login request.
2019-11-13T21:57:56Z [DEBUG] [/core/filter/security.go:442]: can not get user information from session
2019-11-13T21:57:56Z [DEBUG] [/core/filter/security.go:511]: user information is nil

### PR DESCRIPTION
**Pull request by reasonerjt**
This commit fixes #9869
It has some refactor to make sure the group is populated when user is
authenticated via tokenreview workflow.

Signed-off-by: Daniel Jiang <jiangd@vmware.com>

### CODE REVIEW

File: src/pkg/authproxy/http.go [hunk 1]
```diff
@@ -39,27 +43,49 @@ func TokenReview(sessionID string, authProxyConfig *models.HTTPAuthProxy) (*k8s_
...
+	return tokenReviewResponse.Status, nil
+
+}
+
+// UserFromReviewStatus transform a review status to a user model.
+// Group entries will be populated if needed.
+func UserFromReviewStatus(status k8s_api_v1beta1.TokenReviewStatus) (*models.User, error) {
+	if !status.Authenticated {
+		return nil, fmt.Errorf("failed to authenticate the token, error in status: %s", status.Error)
+	}
```
Comment (stonezdj) [inline src/pkg/authproxy/http.go]: In previous implementation, if status.Authenticate = false, it  return false; in current, the error is ignored.

Comment (reasonerjt) [reply src/pkg/authproxy/http.go]: It's mistake, thanks for pointing out!

File: src/core/filter/security.go [hunk 2]
```diff
@@ -346,11 +346,12 @@ func (ap *authProxyReqCtxModifier) Modify(ctx *beegoctx.Context) bool {
 		log.Errorf("User: %s has not been on boarded yet.", rawUserName)
 		return false
 	}
-	if rawUserName != tokenReviewResponse.Status.User.Username {
-		log.Errorf("user name doesn't match with token: %s", rawUserName)
+	u2, err := authproxy.UserFromReviewStatus(tokenReviewStatus)
```
Comment (wy65701436) [inline src/core/filter/security.go]: move this block up to the line 337?

Comment (reasonerjt) [reply src/core/filter/security.go]: move it up make the code a bit hard to read, per discussion, I'll keep it this way.

**Pull request merged**